src/java/org/apache/cassandra/config/CassandraRelevantProperties.java

```diff
@@ -778,6 +778,10 @@ public enum CassandraRelevantProperties
     SAI_VECTOR_USE_PRUNING_DEFAULT("cassandra.sai.jvector.use_pruning_default", "1000"),
     SAI_WRITE_JVECTOR3_FORMAT("cassandra.sai.write_jv3_format", "false"),
 
+    /**
+     * Enable schema backward compatibility mode for CC 4.0.
+     */
+    SCHEMA_BACKWARD_COMPATIBILITY_CC_4("cassandra.schema.backward_compatibility_cc_4", "false"),
```

Comment:
this should be true.
also I think this should just be a general 4.0 backwards compatibility flag, rather than a schema one?  We probably just want the whole thing in backwards compatibility mode?
I think there is actually already a flag in the C* yaml file about storage compatibility that we could probably re-use, rather than adding a new -D.

Reply:
I updated with a new commit that uses `StorageCompatibilityMode` instead. I added a new mode, `CC4`, to distinguish that CC4 supports BTI/TrieIndex, while vanilla `CASSANDRA_4` does not.